test/hotspot/jtreg/runtime/cds/appcds/loaderConstraints/LoaderConstraintsTest.java

```diff
@@ -37,33 +38,68 @@
 import com.sun.net.httpserver.HttpHandler;
 import jdk.test.lib.Asserts;
 import jdk.test.lib.helpers.ClassFileInstaller;
+import jdk.test.lib.Platform;
 
 public class LoaderConstraintsTest  {
     static String mainClass = LoaderConstraintsApp.class.getName();
+    static String httpHandlerClass = HttpHandler.class.getName().replace(".", "/");
+    static String httpExchangeClass = HttpExchange.class.getName().replace(".", "/");
     static String appJar = null;
     static String appClasses[] = {
         mainClass,
-        HttpHandler.class.getName(),
-        HttpExchange.class.getName(),
+        httpHandlerClass,
+        httpExchangeClass,
         Asserts.class.getName(),
```

Comment:
Why other class names don't need to do `replace(".", "/")`?

Reply:
The problem is that the classlist requires the slash separator. I.e., `com/sun/net/httpserver/HttpHandler` must be used. In contrast, `ClassFileInstaller.writeJar()` allows both `jdk.test.lib.Asserts` and `jdk/test/lib/Asserts`. Since the `Asserts` class is used only by `ClassFileInstaller`, I left it as is.

To make it easier to write test cases, I have filed https://bugs.openjdk.java.net/browse/JDK-8267624 "CDS classlist should allow both . and / in class names"